application/src/main/java/org/thingsboard/server/service/edge/rpc/processor/BaseEdgeProcessor.java

```diff
@@ -577,14 +316,14 @@ protected CustomerId safeGetCustomerId(long mSB, long lSB) {
 
     protected boolean isEntityExists(TenantId tenantId, EntityId entityId) {
         return switch (entityId.getEntityType()) {
-            case TENANT -> tenantService.findTenantById(tenantId) != null;
-            case DEVICE -> deviceService.findDeviceById(tenantId, new DeviceId(entityId.getId())) != null;
-            case ASSET -> assetService.findAssetById(tenantId, new AssetId(entityId.getId())) != null;
-            case ENTITY_VIEW -> entityViewService.findEntityViewById(tenantId, new EntityViewId(entityId.getId())) != null;
-            case CUSTOMER -> customerService.findCustomerById(tenantId, new CustomerId(entityId.getId())) != null;
-            case USER -> userService.findUserById(tenantId, new UserId(entityId.getId())) != null;
-            case DASHBOARD -> dashboardService.findDashboardById(tenantId, new DashboardId(entityId.getId())) != null;
-            case EDGE -> edgeService.findEdgeById(tenantId, new EdgeId(entityId.getId())) != null;
+            case TENANT -> edgeCtx.getTenantService().findTenantById(tenantId) != null;
+            case DEVICE -> edgeCtx.getDeviceService().findDeviceById(tenantId, new DeviceId(entityId.getId())) != null;
+            case ASSET -> edgeCtx.getAssetService().findAssetById(tenantId, new AssetId(entityId.getId())) != null;
+            case ENTITY_VIEW -> edgeCtx.getEntityViewService().findEntityViewById(tenantId, new EntityViewId(entityId.getId())) != null;
+            case CUSTOMER -> edgeCtx.getCustomerService().findCustomerById(tenantId, new CustomerId(entityId.getId())) != null;
+            case USER -> edgeCtx.getUserService().findUserById(tenantId, new UserId(entityId.getId())) != null;
+            case DASHBOARD -> edgeCtx.getDashboardService().findDashboardById(tenantId, new DashboardId(entityId.getId())) != null;
+            case EDGE -> edgeCtx.getEdgeService().findEdgeById(tenantId, new EdgeId(entityId.getId())) != null;
```

Comment:
This is pretty inefficient, do some refactoring please :)
Use `EntityDaoRegistry.getDao(entityId.getEntityType).existsById`.